bytestring_splitter/__init__.py

```diff
@@ -44,18 +48,33 @@ def set_original_bytes_repr(self, bytes_representation):
         self._original_bytes = bytes_representation
 
     def finish(self):
-        finished_values = {}
+
         for message_name, (bytes_for_message, message_class, kwargs) in self.processed_objects.items():
-            finished_values[message_name] = produce_value(message_class,
+            self._finished_values[message_name] = produce_value(message_class,
                                                           message_name,
                                                           bytes_for_message,
                                                           kwargs)
-        return self._receiver(**finished_values, **self._additional_kwargs)
+        return self._receiver(**self._finished_values, **self._additional_kwargs)
+
+    def __getattr__(self, message_name):
+        # First we'll try to see if this message_name already has a finished value:
+        try:
+            return self._finished_values[message_name]
+        except KeyError:  # suppress might be good here, but it appears to have a performance penalty, and this is a performance-concerned function.
+            pass
```

Comment:
Heh - interesting - does `.get` share a similar performance hit?

Reply:
`.get` has what appears to be a measurable, but substantially smaller performance hit.  Here's [what I said in Discord](https://ptb.discordapp.com/channels/411401661714792449/411401661714792451/698011920736976937) about it:

```
Interesting: I just ran three test scenarios on the "just in time attribute access", each access an attribute 500000 times from a partially split bytestring.

It's possible that the value has been "finished" (ie, passed to its receiver), so we have to both check that and return the finished value, and then finish the value if it hasn't previously been.

They are both dicts.

I tried with .get(...), with a try block (and except: pass), and suppress.

The times were more different than I thought:

The try block was about 4.2 seconds, the .get was about 4.6 seconds, and the suppress was 6.2 seconds (!).
```